sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+
+/**
+ * @file az_iot_pnp_client.h
+ *
+ * @brief definition for the Azure IoT PnP device SDK.
+ */
+
+#ifndef _az_IOT_PNP_CLIENT_H
+#define _az_IOT_PNP_CLIENT_H
+
+#include <az_result.h>
+#include <az_span.h>
+
+// NOTE: az_iot_hub_client.h is included for internal use only.
+#include <az_iot_hub_client.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * @brief Azure IoT PnP Client options.
+ *
+ */
+typedef struct az_iot_pnp_client_options
+{
+  az_span user_agent; /**< The user-agent is a formatted string that will be used for Azure IoT
+                         usage statistics. */
+} az_iot_pnp_client_options;
+
+/**
+ * @brief Azure IoT PnP Client.
+ *
+ */
+typedef struct az_iot_pnp_client
+{
+  struct
+  {
+    az_iot_hub_client iot_hub_client;
+	az_span root_interface_name;
+  } _internal;
+} az_iot_pnp_client;
+
+/**
+ * @brief Gets the default Azure IoT PnP Client options.
+ * @details Call this to obtain an initialized #az_iot_pnp_client_options structure that can be
+ *          afterwards modified and passed to #az_iot_pnp_client_init.
+ *
+ * @return #az_iot_pnp_client_options.
+ */
+AZ_NODISCARD az_iot_pnp_client_options az_iot_pnp_client_options_default();
+
+/**
+ * @brief Initializes an Azure IoT PnP Client.
+ *
+ * @param[out] client The #az_iot_pnp_client to use for this call.
+ * @param[in] iot_hub_hostname The IoT Hub Hostname.
+ * @param[in] device_id The Device ID.
+ * @param[in] root_interface_name The root interface of the #az_iot_pnp_client.
+ * @param[in] options A reference to an #az_iot_pnp_client_options structure. Can be NULL.
+ * @return #az_result.
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_init(
+    az_iot_pnp_client* client,
+    az_span iot_hub_hostname,
+    az_span device_id,
+    az_span root_interface_name,
+    az_iot_pnp_client_options const* options);
+
+/**
+ * @brief Gets the MQTT user name.
+ *
+ * The user name will be of the following format:
+ * {iothubhostname}/{device_id}/?api-version=2018-06-30&{user_agent}?digital-twin-model-id={root_interface_name}
```

Comment:
should the character after the `{user_agent}` be a `&` instead of a `?`

Reply:
Yup!